### ISSUE TITLE
Save intermediate data products

### ISSUE THREAD

**Issue by virajkaram** [winter-telescope/mirar#161]
It will be useful to save not just the final image but some intermediate data products. 
I think the results of the following steps will be useful to save - 
1. Detrended (flat+bias corrected)
2. Astrometry on detrended image (astrometry just adds to the header, so we can just overwrite the detrended images with an updated header whenever astrometry runs successfully)
3. Resampled images (swarp affects the data so this needs to be its own image)
4. Final photometrically calibrated image (photcal also only updates the header, so we can overwrite the resampled image). We currently save only this image. 

In short, save one more image than what we are currently doing.

**Comment by robertdstein** [comment]
I have three comments on this:

* I really do not like the idea of having an image overwritten some of the time. I would much prefer to not have 2 and 4 overwriting 1 and 3. I think it would really lead to a lot of confusion in the long run, if you pull up image xyz from night abc, and need to consult either header or database to work out what stage of the pipeline it corresponds to. For the same night, multiple images in the same folder (requested the same way) would correspond to different stages of processing. 
* Saving N times as many additional images increases the data storage requirement by factor N. So in this proposal we are effectively doubling the volume of data from Winter. So the change comes with a cost.
* I think we should think carefully about what the actual use case for these intermediate images, and whether they outweigh the cost of saving them. I think they are basically only needed for rare cases where processing fails for a particular subset of images AND somebody (e.g us) wants to debug things by hand. However, instead of saving these many images from the first run of the pipeline, we could think of ways to easily rerun processing for specific images. For example a simple "debug mode" saving these intermediate steps to be run locally, or even to run by request via a web form. My gut feeling is that would be a more proportionate solution, but I think we should discuss it in any case.
